File: freshmart/src/types/notification.ts
export type NotificationType = 'warning' | 'info' | 'success' | 'error';

export interface NotificationItem {
  id: string;
  title: string;
  message: string;
  time: string;
  type: NotificationType;
  read: boolean;
  isRead?: boolean;
  createdAt?: string;
  employeeId?: string;
  employeeName?: string;
}

export interface CreateNotificationRequest {
  title: string;
  message: string;
  type: NotificationType;
  employeeId?: string;
}

export interface UnreadCountResponse {
  unreadCount: number;
}

File: Models/Notification.cs
using System;
using dotnet_backend_freshmart.Models.Enums;

namespace dotnet_backend_freshmart.Models
{
    public class Notification
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        public string Title { get; set; } = string.Empty;

        public string Message { get; set; } = string.Empty;

        public NotificationType Type { get; set; } = NotificationType.Info;

        public bool IsRead { get; set; } = false;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public Guid? EmployeeId { get; set; }

        public Employee? Employee { get; set; }
    }
}
